ocean_lib/assets/test/test_asset_downloader.py

```diff
@@ -5,98 +5,139 @@
 
 import pytest
+
 from ocean_lib.agreements.service_types import ServiceTypes
+from ocean_lib.assets.asset import Asset
 from ocean_lib.assets.asset_downloader import download_asset_files
 from ocean_lib.data_provider.data_service_provider import DataServiceProvider
+from ocean_lib.web3_internal.currency import to_wei
 from tests.resources.ddo_helpers import get_sample_ddo
+from tests.resources.helper_functions import deploy_erc721_erc20, create_basics
 
 
-def test_ocean_assets_download_failure(publisher_wallet):
+def test_ocean_assets_download_failure(publisher_wallet, config):
     """Tests that downloading from an empty service raises an AssertionError."""
-    data_provider = DataServiceProvider
 
-    ddo = get_sample_ddo()
-    sa = ddo.get_service(ServiceTypes.ASSET_ACCESS)
-    sa.service_endpoint = None
-    ddo.services[1] = sa
+    ddo_dict = get_sample_ddo()
+    ddo = Asset.from_dict(ddo_dict)
+    access_service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    access_service.service_endpoint = None
+    ddo.services[0] = access_service
 
     with pytest.raises(AssertionError):
         download_asset_files(
-            sa.index,
             ddo,
+            config.provider_url,
             publisher_wallet,
             "test_destination",
-            "",
             "test_order_tx_id",
-            data_provider,
         )
 
 
-def test_ocean_assets_download_indexes(publisher_wallet):
+def test_invalid_provider_uri(publisher_wallet):
+    """Tests with invalid provider URI that raise AssertionError."""
+    provider_uri = "http://mock/"
+    ddo_dict = get_sample_ddo()
+    ddo = Asset.from_dict(ddo_dict)
+
+    with pytest.raises(AssertionError):
+        download_asset_files(
+            ddo, provider_uri, publisher_wallet, "test_destination", "test_order_tx_id"
+        )
+
+
+def test_ocean_assets_download_indexes(publisher_wallet, config):
     """Tests different values of indexes that raise AssertionError."""
-    data_provider = DataServiceProvider
 
-    ddo = get_sample_ddo()
-    sa = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    ddo_dict = get_sample_ddo()
+    ddo = Asset.from_dict(ddo_dict)
 
     index = range(3)
     with pytest.raises(TypeError):
         download_asset_files(
-            sa.index,
             ddo,
+            config.provider_url,
             publisher_wallet,
             "test_destination",
-            "",
             "test_order_tx_id",
-            data_provider,
-            index,
+            index=index,
         )
 
     index = -1
     with pytest.raises(AssertionError):
         download_asset_files(
-            sa.index,
             ddo,
+            config.provider_url,
             publisher_wallet,
             "test_destination",
-            "",
             "test_order_tx_id",
-            data_provider,
-            index,
+            index=index,
         )
+
     index = 4
-    with pytest.raises(AssertionError):
+    with pytest.raises(Exception):
```

Comment:
we never assert raising a plain Exception. If this Exception comes from provider, make sure you catch it and rethrow an AssertionError instead.